src/base/bittorrent/session.cpp

```diff
@@ -2149,11 +2165,31 @@ bool Session::addTorrent_impl(const std::variant<MagnetUri, TorrentInfo> &source
                 loadTorrentParams.name = contentName;
         }
 
+        Q_ASSERT(addTorrentParams.filePaths.isEmpty() || (addTorrentParams.filePaths.size() == metadata.filesCount()));
+
+        const TorrentContentLayout contentLayout = ((loadTorrentParams.contentLayout == TorrentContentLayout::Original)
+                                                    ? getContentLayout(metadata.filePaths()) : loadTorrentParams.contentLayout);
+        const QStringList filePaths = toContentLayout(
```

Comment:
Minor note, I had a little trouble locating `toContentLayout()`. I think it would be better if it had been under its own namespace.

Reply:
> I had a little trouble locating `toContentLayout()`.

It is located alongside with `TorrentContentLayout` declaration (inside `BitTorrent`) namespace. Maybe it isn't so good but I can't imagine any better location. Can you suggest something?

Reply:
Nevermind, lets stay as is for now.